### DISCUSSION TITLE
Recent activities are gone from the top panel

### DISCUSSION THREAD

**Discussion by asaunin** [community/community#4227] [Other Feature Feedback, Questions, & Ideas]
Does anyone know, why recent activity panel was removed? 
It was an extremely useful feature! I've never used PR menu, and managed everything with Recent activities instead. 

It contained all the necessary information in a single place (created, assigned, mentioned, commented but not fully reviewed PR's assigned to me).

![image](https://user-images.githubusercontent.com/19559375/122787527-1867b100-d2be-11eb-9697-3987eb430e72.png)

**Comment by Arhell** [comment]
Agree with the author

Take back, or give a choice.
Everything was at the top and in front of your eyes and you could easily follow the changes, now you need to move your eyes to the lower left corner, and if there are a lot of repositories, you can also scroll a lot.

**Reply by Andre601** [reply]
Not sure what screen dimensions you have, but for me is there not so many repos to not see the list.
![image](https://user-images.githubusercontent.com/11576465/123332052-4eb46300-d540-11eb-8822-825aed5b0ba5.png)

**Reply by mbigras** [reply]
@Arhell I totally agree that the activity feed should be more of an activity feed and less of a news feed. I care more about the pull requests that I'm working on with others than the material that's in the front-and-center _All activity_ column. I would rather see the pull requests I've opened or reviewed in that front-and-center column.

My usual workflow is:

- Go to github.com
- Look at which pull requests have been reviewed
- Incorporate changes and do work
- Go to github.com
- Look at which pull requests have been reviewed

Now I'm not sure what status of my pull requests is unless I highlight my mouse over each small line item in the **Recent activity** side menu.

**Reply by Buanter** [reply]
Yah,I'am sory brother,,,my it's lemod by star licencing my had

**Comment by Andre601** [comment]
The panel wasn't removed.

It was just put on the left where the repository and team lists are.

**Reply by Buanter** [reply]
Y I'no dad,it's really in my life the work is fana men

**Comment by Erol444** [comment]
It's way worse than it used to be. Please bring it back or make it customizable.

**Reply by Buanter** [reply]
Thanks brother,I'am no perfect it's people

**Comment by bdresser** [comment]
Hi folks, thanks for your feedback. 

We moved the Recent Activity component to create space for improvements we're making to the Activity Feed. We hear your feedback and would love to understand more about your usecase and workflows.

* Do you use notifications on GitHub (https://github.com/notifications)? If so, how do you use recent activity differently from notifications? If you don't use notifications, why not? 
* Are most of the contents of recent activity from the same repository, or spread out across several different repositories? 
* Do you use the homepage for anything other than recent activity – like the sidebar repo links, or scrolling the central feed? 
* What's specifically missing from the updated version of recent activity that you found useful? Comment count, avatars, full titles, etc?

**Reply by Andre601** [reply]
- I for my part use notification sparingly. At most I go there, select all entries and mark them as done. I rely much more on e-mail notifications as they are somewhat easier to figure out in terms of what has changed (i.e. what is the comment)
- It's a mixed bag in my case. Having recent activity about opened PRs, comments on such or on issues and similar. Maybe include discussions to this list? Or perhaps even have a separate list for discussions? Because I think that's something worth to be added.
- Yes. I often use the repos listed since they're often the ones I use the most so it's quick and easy for me to get.
- I for my part don't hate or even miss the previous design of recent activity as I already used it barely. But to be fair does it feel a bit... blant by just showing (part of) the title and not any more info such as why is it now displayed at the top of the list.

**Reply by asaunin** [reply]
Hi @bdresser!

> Do you use notifications on GitHub (https://github.com/notifications)? If so, how do you use recent activity differently from notifications? If you don't use notifications, why not?

Any type of notifications are disabled by default, and I'm not going to enable them. I prefer to request updates/changes, when I have time to handle em, instead of being interrupted by push notifications.

> Are most of the contents of recent activity from the same repository, or spread out across several different repositories?

Different ones across the company

> Do you use the homepage for anything other than recent activity – like the sidebar repo links, or scrolling the central feed?

Search and recent activity. Other panels I don't use at all in my regular work. 
I assume that `Issues` might be useful for other people, but in my particular case, we track everything in Jira.

> What's specifically missing from the updated version of recent activity that you found useful? Comment count, avatars, full titles, etc?

Everything was useful there, cause it gave me all the information in a small block:
- who prepared PR and when;
- is it approved or not;
- who was reviewing it;
- are there any comments or not;

But the main advantage, prom my prospective, is that recent activity collected at the same time in one place both PR created by me, PRs in review state and PR's assigned to me.

**Reply by 2Pacalypse-** [reply]
In my case, the benefits of the previous version of the recent activity design were the following:
- It showed all the relevant information in top/center of the homepage, allowing for a very quick navigation to all of the points of interest (e.g. issues/PRs created by me, issues assigned to me, and PRs requesting my review).
- It showed all of the above things even if you had a repository fully ignored (which I usually did for high frequency repositories, so I don't get spammed by their emails).
- It wasn't distracting. I.e. there weren't any blinking lights or whatever indicators are used nowadays to get your attention that something changed in the system. All it did was put the most recent activity on top, iirc.
- There was a "Show more" button that allowed you to see all your activities (I assume? I never had some crazy numbers there)

Now, the current activity list is pretty much useless in those regards. Not only does it show only 4 activities, the names are so short and lack any information that you pretty much _have_ to hover on each one to get any kind of information (at least that won't take long since there are only 4!). On the other hand, the notifications page seem like a most promising successor to the recent activity list, but it's lacking in certain areas to make them fully replace the old version of recent activity list to me:
- They're hidden behind a separate page.
- It doesn't show notifications from ignored repositories (this is understandable, and would be fine if it were possible to configure email notifications per-repository. Maybe it is?).
- The new notification indicator can be pretty distracting, leading to either immediately doing whatever's needed to clear it without actually handling whatever caused the notification (what I do), or start ignoring it until you have actual time to deal with it and ending up with pretty much a permanent notification indicator (what my friend does).
- Minor nitpick: "Mark as read" action is hidden behind a menu for some reason while there is plenty of space beside the "Done" and "Unsubscribe" buttons.
- Notification filters should apply to whatever notifications are currently being listed. From what I can tell, they're not applied to the "Done" category.

Regardless of all of the above, I feel like notifications are pretty close to being a full replacement to the recent activity list (and several other GitHub features). My suggestion would be to move the notifications page to the front page. Have the filter name/icon visually shown next to each notification, perhaps even allow grouping/ordering the notifications by filters. This way users would be able to have things they have active interest in at the top of the page (like created PRs, assigned issues/reviews, etc.). Then below that they could see stuff they have passive interest in, e.g. subscribed threads, social notifications, etc. And finally at the end they can have an info dump of all of the activity (commits from various repositories), which is currently the most visible thing at the front page.

And to cap off this very short post, here are my answers to your direct questions:
>Do you use notifications on GitHub (https://github.com/notifications)? If so, how do you use recent activity differently from notifications? If you don't use notifications, why not?

I answered most of this above, but a short recap: I use notifications, mainly to clear them so I don't see the unread indicator anymore. The difference between notifications and recent activity that was crucial to me was that recent activity showed the activities from ignored repositories and notifications do not.
>Are most of the contents of recent activity from the same repository, or spread out across several different repositories?

Spread out across several different repositories.
>Do you use the homepage for anything other than recent activity – like the sidebar repo links, or scrolling the central feed?

I use sidebar repo links and that's it.
>What's specifically missing from the updated version of recent activity that you found useful? Comment count, avatars, full titles, etc?

This was also answered above, but pretty much everything. The current version of the recent activity lacks any information while unhovered that is pretty much useless.

**Reply by ericoporto** [reply]
@bdresser (more here: https://github.community/t/how-to-roll-to-previous-recent-activity-theme/185688)

> Do you use notifications on GitHub (https://github.com/notifications)? If so, how do you use recent activity differently from notifications? If you don't use notifications, why not?

I don't use notifications, the blue dot gives me anxiety and I use GitHub to browse historical codebases that are valuable when I am fixing a bug. If the blue dot (that plagues ALL PAGES) comes up I get distracted. I turned it off.

> Are most of the contents of recent activity from the same repository, or spread out across several different repositories?

Several repositories, usually I go to a bug, findout the things is upstream, then go to the other repo and do a PR or open an issue to fix the bug that is showing up on my project.

> Do you use the homepage for anything other than recent activity – like the sidebar repo links, or scrolling the central feed?
What's specifically missing from the updated version of recent activity that you found useful? Comment count, avatars, full titles, etc?

Nope. Never. I only use the feed for the organizations I participate, the personal feed is completely useless.


Additionally, I mostly use the GitHub interface from my phone, when I am in the middle of other things, triaging bugs and other things. The current recent activity limited to 4 things and without extra information reaaaally is terrible for me.

**Reply by hst-pierre** [reply]
@bdresser 
* We don't as we are using Slack to get notifications on any changes with the integration
* Spread across several repositories
* I often use the sidebar repo links since that's basically all projects I work with (and the list grow nicely). I am not using the central feed since it's currently empty even if I watch or star some repos.
* The most problematic issue I have right now is that its only four lines and it cannot grow like it used to be. Four is very few given I contribute to several repositories (so multiple PR open at the same time), not counting all other teams asking me to review their PRs. Next for me would be avatar as this was a great indicator for me that I already participated in a PR (so most likely approved it) and any PR not having my face (except mine), was telling me which PR needed my approval. I could live with a hovered title even though the first part of the title is often useless to me (we use semantic releases so title often starts with type and scope, following by the actual change).

**Reply by sinanislekdemir** [reply]
Hi @bdresser 

* The organization I work for has over 100 repositories, also there are other projects that I am watching. So, notifications is generally a mess for me.
* Most contents of recent activity are from over 4 repositories for sure. Left sidebar recent activity just shows the latest 4, which is a bit misleading in my case.
* I use sidebar to fast-access to the last repositories I've contributed to.
* With the existing state, it's not clear which repository it is. Also, long PR titles are shortened. Also, I can't see the last commenter or when the last change/comment was. I don't care much about the avatar tho.

As an alternative to the missing information, I tried using the Pull Requests section (https://github.com/pulls) but somehow that doesn't give relevant information. I usually can't see any PR's I am assigned to within the private repositories of the organization.

I am really missing the old recent activity dashboard 😢 

> Hi folks, thanks for your feedback.
> 
> We moved the Recent Activity component to create space for improvements we're making to the Activity Feed. We hear your feedback and would love to understand more about your usecase and workflows.
> 
>     * Do you use notifications on GitHub (https://github.com/notifications)? If so, how do you use recent activity differently from notifications? If you don't use notifications, why not?
> 
>     * Are most of the contents of recent activity from the same repository, or spread out across several different repositories?
> 
>     * Do you use the homepage for anything other than recent activity – like the sidebar repo links, or scrolling the central feed?
> 
>     * What's specifically missing from the updated version of recent activity that you found useful? Comment count, avatars, full titles, etc?

**Reply by bdresser** [reply]
Thank you for the detailed feedback, folks. We'll take this back to the team. Thank you for your patience!

**Reply by mbigras** [reply]
@bdresser Thank you for giving this issue some attention and gathering feedback. What is the status of the Recent activity pane? That pane summarized the pull requests that need attention and was very useful. The Recent activity pane served me as a daily tool for years and now has been overshadowed by the All activity pane which is spam that I can't filter or configure. I'll add my answers to your questions but the summary is: The Recent activity pane was very helpful and I wish I could use it again.

>Do you use notifications on GitHub (https://github.com/notifications)? If so, how do you use recent activity differently from notifications? If you don't use notifications, why not?

I don't use notifications. They don't summarize which pull requests I reviewed and which I have not. They don't automatically clear when a pull request is merged. They aren't configurable and are noise in the form of a blue orb that also isn't configurable. Furthermore they add to the pool of notifications that distract from deep work and I wish I could turn them off.

>Are most of the contents of recent activity from the same repository, or spread out across several different repositories?

Recent activity is spread across multiple repositories and that's why the summary view is useful.

>Do you use the homepage for anything other than recent activity – like the sidebar repo links, or scrolling the central feed?

I only use the homepage for the recent activity pane because it summarizes the work that needs my attention.

>What's specifically missing from the updated version of recent activity that you found useful? Comment count, avatars, full titles, etc?

The following information was what I used:

* Which repository does this pull request come from?
* What is the full title of the pull request?
* Has a review or comment happened that needs my attention?

Now all the that information is hidden or truncated.

**Comment by lewislbr** [comment]
Also worth mentioning: the forced display of all starred repos updates in All Activity. IIRC previously you could choose which repos you wanted to see updates from with the specific "watch" button. What does the "watch" button do now? I guess the only option to not see updates from starred repos is to unstar them? That'd be a bit unfair.

**Comment by jogibear9988** [comment]
I also don't use "notifications". In notifications there are infos from many repositorys, I don't work on. Maybe cause they are in my organisation.

I only wanted to see, changes to issues I created, I commented, .... => The view Recent Activity gave me. It would be even more better if I could scroll in more then the last 12 items (ok at the moment only 4).

The Docu now is also wrong, see: https://docs.github.com/en/account-and-profile/setting-up-and-managing-your-github-user-account/managing-user-account-settings/about-your-personal-dashboard#finding-your-recent-activity

@bdresser 
any state of the current decission?

**Comment by mbigras** [comment]
For a workaround I don't use the homepage anymore. Instead I bookmarked this page: https://github.com/pulls/review-requested.

**Comment by mszhanyi** [comment]
No, I don't like notification yet. There're too many noises and I need to create all kinds of rules.
Especially, when I'm back from a long vacation, there're too many emails.
I'm a lazy man, I'd like to see more activities involved me in Github at one glance.

Generally, `All activity` is useless for me though I subscribed some repos.

**Comment by ghost** [comment]
I only able to see the last 4 recent activities on the left.
How to watch all the activities?

**Comment by HoldOffHunger** [comment]
This is absolutely disgusting.  As of yesterday, all activity-checking from landing pages has been completely removed and replaced with advertisements for paying repos, like Laravel.  I don't give a crap about Laravel, get the hell out of here, you M$ androids!

**Reply by Andre601** [reply]
I can't see what you're talking about.

Left side still shows my recent activity... Not some kind of ad or anything.

**Comment by HoldOffHunger** [comment]
Hrm, nope, just getting random people listed I'm not following in my feed:

https://i.imgur.com/PQiwZlj.png

**Reply by Andre601** [reply]
This is the "For you" feed from GitHub.
It's not the recent activities you yourself made.
![grafik](https://user-images.githubusercontent.com/11576465/198418613-913349f0-1c51-4889-a3aa-2a9e17c6f37a.png)

Not sure why you're not having the normal feed tab there, so maybe check if you're using an extension that may alter this somehow...

**Comment by HoldOffHunger** [comment]
>It's not the recent activities you yourself made.

It's the default page that I see when I load `github.com`.  Without any modifications from myself, whatsoever.

"Made for myself."  I like that.  Gaslighting.  Thanks.

**Comment by HoldOffHunger** [comment]
Holy shit, GitHub just rolled back their garbage changes.  Rock on.  Great job, everyone.

**Reply by Andre601** [reply]
I mean... If you consider *this* better...?

![grafik](https://user-images.githubusercontent.com/11576465/201136583-08280989-78d2-4754-95f8-3599b76af475.png)
